### ISSUE TITLE
SBApplication.FromBundleIdentifier<T> unexpected NullReferenceException for unknown bundle

### ISSUE THREAD

**Issue by snechaev** [xamarin/xamarin-macios#9011]
### Steps to Reproduce

1. Build and run code attached below
2. Observe NullReferenceException in the last line in Main.

### Expected Behavior
`SBApplication.FromBundleIdentifier<T>(string bundleId)` in case when when `bundleId` is unknown, should return null to be consistent with non-generic version of `SBApplication.FromBundleIdentifier`.

### Actual Behavior
`SBApplication.FromBundleIdentifier<T>` throws NullReferenceException in case when when bundle id is unknown.

### Environment

<details>
  <summary>Environment details</summary>

```
=== Visual Studio Professional 2019 for Mac ===

Version 8.6.5 (build 23)
Installation UUID: 2cba28b3-64e9-4d23-a8c9-21683ba870db
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 610000104

=== Mono Framework MDK ===

Runtime:
	Mono 6.10.0.104 (2019-12/5d03a6fe116) (64-bit)
	Package version: 610000104

=== Roslyn (Language Service) ===

3.6.0-3.20210.9+4eafdcb1bcbd8d3573f2ba6065e56d9b9ce4f8a3

=== NuGet ===

Version: 5.6.0.6591

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/3.1.301/Sdks
SDK Versions:
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.10.0/lib/mono/msbuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14

=== Xamarin.Profiler ===

Version: 1.6.12.29
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.5 (16139)
Build 11E608c

=== Xamarin.Mac ===

Version: 6.18.2.1 (Visual Studio Professional)
Hash: 29c4ea731
Branch: d16-6
Build date: 2020-05-26 17:03:04-0400

=== Xamarin.iOS ===

Version: 13.18.2.1 (Visual Studio Professional)
Hash: 29c4ea731
Branch: d16-6
Build date: 2020-05-26 17:03:05-0400

=== Xamarin Designer ===

Version: 16.6.0.329
Hash: d4f8bcd13
Branch: remotes/origin/d16-6
Build date: 2020-04-24 02:16:02 UTC

=== Xamarin.Android ===

Not Installed

=== Microsoft OpenJDK for Mobile ===

Java SDK: Not Found

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.6.0.50
Hash: 5901879
Branch: remotes/origin/d16-6
Build date: 2020-06-10 22:42:50 UTC

=== Android Device Manager ===

Version: 16.6.0.96
Hash: 6e8b80b
Branch: remotes/origin/d16-6
Build date: 2020-06-10 22:43:28 UTC

=== Build Information ===

Release ID: 806050023
Git revision: 5289d413b99fddfc20b4ecf3e445ccb822213427
Build date: 2020-06-18 12:08:30-04
Build branch: release-8.6
Xamarin extensions: 5289d413b99fddfc20b4ecf3e445ccb822213427

=== Operating System ===

Mac OS X 10.15.5
Darwin 19.5.0 Darwin Kernel Version 19.5.0
    Tue May 26 20:41:44 PDT 2020
    root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

FileNesting 0.1.2
Open With 0.2


```
</details>

### Build Logs

<details>
<summary>log</summary>
Rebuilding...

Cleaning solution sbNRE (Debug)
Build started 06.07.2020 14:14:13.
__________________________________________________
Project "/Users/sergey/work/sbNRE/sbNRE/sbNRE.csproj" (Clean target(s)):

Target CoreClean:
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/bin/Debug/sbNRE.exe".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/bin/Debug/sbNRE.pdb".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/actool/_PartialAppManifest.items".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/actool/_BundleResourceWithLogicalName.items".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/sbNRE.csprojAssemblyReference.cache".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/sbNRE.csproj.CoreCompileInputs.cache".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/sbNRE.exe".
    Deleting file "/Users/sergey/work/sbNRE/sbNRE/obj/Debug/sbNRE.pdb".
Target _CleanBindingResourcePackage:
    Directory "/Users/sergey/work/sbNRE/sbNRE/bin/Debug/sbNRE.resources" doesn't exist. Skipping.
Target _CleanAppBundle:
    Removing directory "bin/Debug/sbNRE.app/".
Target _CleanIntermediateToolOutput:
    Removing directory "obj/Debug/actool".
    Removing directory "obj/Debug/ibtool".
    Directory "obj/Debug/metal" doesn't exist. Skipping.
    Directory "obj/Debug/scntool" doesn't exist. Skipping.
    Directory "obj/Debug/TextureAtlas" doesn't exist. Skipping.
    Removing directory "obj/Debug/".

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.13

Building solution sbNRE (Debug)
Build started 06.07.2020 14:14:14.

</details>

### Example Project (If Possible)

[sbNRE.zip](https://github.com/xamarin/xamarin-macios/files/4877309/sbNRE.zip)

**Comment by chamons** [comment]
I can confirm from code inspection:

```
			using (var u = FromBundleIdentifier (ident))
				return (T)System.Activator.CreateInstance (typeof(T), BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance, null, new object [] { u.Handle }, null);

```

We likely need to add an if null check in the three using blocks in SBApplication.cs

**Comment by whitneyschmidt** [comment]
Fixed in the associated PR #9620
